docs/Security_Enhancement_For_AI_Models.md

```diff
@@ -0,0 +1,246 @@
+# Security Enhancement For AI Models on Cloud Native Environment
+
+AI models often rely on large and sensitive datasets for training, and sometimes can be considered valuable intellectual property. Protecting the AI models is essential to prevent data breaches and unauthorized access to sensitive information. Typically, a common way to protect AI models is encryption, and some encryption algorithms (such as AES GCM) can ensure the confidentiality and integrity of the data.
+
+However, although a model has been encrypted, it would be loaded after decryption, and the runtime environment is untrusted most times, especially for cloud native environments. This document describes a design and implementation to enhance the security of the flow in a Trusted Execution Environment (TEE).
+
+## 1. Design
+
+![](secure-model-design.png)
+
+
+### 1.1 Model Preprocess
+
+The preprocess to protect the model and the encryption key include:
+
+- Generate a key, a AES GCM key can be used for confidentiality and integrity.
+- Encrypt the model by the key.
+- Upload the encypted model to the model server.
+- Register key to Key Broker Service (KBS), and KBS communicates with key management service (KMS) to store the key.
+
+
+### 1.2 Confidential Cloud Native Primitives (CCNP)
+
+[CCNP](https://github.com/intel/confidential-cloud-native-primitives)  is the solution targeted on simplifying the use of Trusted Execution Environment (TEE) in cloud-native environment. There are 2 parts included in CCNP, the services and the SDK.
```

Comment:
extra space between CCNP and `is`

Reply:
change the latest sentence to `It comprises two main components: the services and the SDK.`

Reply:
Done.